lib/index.ts

```diff
@@ -38,101 +38,133 @@ export function main (): void {
 
         files.forEach(function (fileName) {
             if(fileName.startsWith('.')) return;
-            if(!fileName.endsWith('.json')) return;
+            if(!fileName.endsWith('.json')) {
+                console.log('Ignoring "' + fileName + '".');
+                return;
+            }
 
             const rawJSON = fs.readFileSync(path.join(sourcePath, fileName));
             const parsedJSON = JSON.parse(rawJSON.toString().replace(/\\/g, '\\\\'));
             if (typed === null)
-                typed = buildTypes(parsedJSON);
+                typed = buildHelpers(parsedJSON);
             buildLang(fileName, parsedJSON);
         });
     });
 }
 
-function die (explanation: string): void {
+function die (explanation: string): never {
     console.log(explanation);
-    process.exit(1);
+    return process.exit(1);
 }
 
-function buildTypes (data: JSON): boolean {
-    console.log('Bulding Types.elm');
-    const filePath = path.join(destNamespacePath, 'Types.elm');
-
-    const subprocess = child_process.spawn('elm-format',
-        ['--stdin', '--output', filePath],
-        { stdio: [ 'pipe', 1, 2 ] }
-    );
-
-    if (subprocess.stdin === null) {
-        die('Unable to pipe to elm-format!'); 
-        return false;
-    }
-
-    const buffer: Writable = subprocess.stdin;
-    buffer.write(`module ${moduleNamespace}.Types exposing (..)\n\n\n`);
-
-    addRecord('', data, buffer);
-
-    buffer.end();
+function buildHelpers (data: JSON): boolean {
+    console.log('Bulding Types.elm / Decoders.elm / MockLanguage.elm');
+    const typesBuffer = pipeToElmFormat(path.join(destNamespacePath, 'Types.elm'));
+    const decodersBuffer = pipeToElmFormat(path.join(destNamespacePath, 'Decoders.elm'));
+    const mockBuffer = pipeToElmFormat(path.join(destNamespacePath, 'MockLanguage.elm'));
+
+    typesBuffer.write(`module ${moduleNamespace}.Types exposing (..)\n`);
+    decodersBuffer.write(`module ${moduleNamespace}.Decoders exposing (..)\n`
+        +`import ${moduleNamespace}.Types as Types\n`
+        +'import Json.Decode as Decode exposing (Decoder)\n'
+        +'\n'
+        +decodersHelpers
+        );
+    mockBuffer.write(`module ${moduleNamespace}.MockLanguage exposing (..)\n`
+        +`import ${moduleNamespace}.Types as Types\n`
+        );
```

Comment:
if you're using template strings already no need to manually break lines and append them together

Reply:
:sweat_smile: it's either { this and keep code aligned } or { One big template string and align imports at column 0 }